### ISSUE TITLE
Compatibility with ModSecurity version 2.9.1

### ISSUE THREAD

**Issue by zimmerle** [bitsofinfo/logstash-modsecurity#34]
Hi Guys,

I am following the project for a while, congrats!

Yesterday we have released ModSecurity version 2.9.1. There was a modification in the logs that might affect you. The issue SpiderLabs/ModSecurity#840 contains more details about it.

Also, in v2.9.1 there is this possibility to save the audit logs in JSON format, not sure if the format is structured in the shape that you need for elasticsearch, but may be useful.

**Comment by bitsofinfo** [comment]
Does this affect the audit logs?

**Comment by bitsofinfo** [comment]
I just don't have the time to look at this now @zimmerie, thanks for bringing it to the projects attention though! Perhaps yourself or some prior contributors @MWilkinson @breml could take a look?

**Comment by breml** [comment]
I quickly scrolled through https://github.com/SpiderLabs/ModSecurity/pull/840 and as far as I understand, this affects mainly error.log and not audit.log format. @zimmerle can you confirm?

**Comment by zimmerle** [comment]
Hi @breml,

The only change that seems relevant to you is the fact that now we have this new line:

```
Apache-Error: [file "apache2_util.c"] [line 271] [level 3] [client %s] ModSecurity: %s%s [uri "%s"]%s
```

This happens to appear on the "section H" of the audit logs.

**Comment by Alex-Kw** [comment]
I believe I'm seeing this issue. I'm noticing this issue with section H parsing above 2.9.0. Field names like  auditLogTrailer.Action are normal but there is a field name called "
auditLogTrailer. [line 271] [level 3] [client $IPADDR] ModSecurity" with data starting as the log continues, "Access denied with code ....."

**Comment by bitsofinfo** [comment]
Please submit a PR if you have a patch

**Comment by Alex-Kw** [comment]
If I figure out a good way to handle it, I will. Thank you.

**Comment by Alex-Kw** [comment]
I've come at this a couple ways. In 2081_filter_section_h_convert_to_key-value.conf this was my first solution:

`    if [rawSectionH] =~ /.+/ {
      mutate {
        gsub => [ "rawSectionH", 'ModSecurity:', 'ModSecurity' ]
      }
      kv {
        source => "rawSectionH"
        field_split => "\n"
        value_split => ":"
        target => "auditLogTrailer"
      }
`

This removed the extra colon but some things like REQUEST_HEADERS:User_Agent were still problematic.

I ended up doing this dirty hack to just drop the new "Apache-Error" line from raw section H since all the data in it was redundant for me. I'm not sure if there is really any need for it given the pieces of info are already extracted by your script(s) otherwise.

`
  if [rawSectionH] =~ /.+/ {
    mutate {
    #   gsub => [ "rawSectionH", 'ModSecurity:', 'ModSecurity' ]
        gsub => [ "rawSectionH", '^Apache-Error:.*', '' ]
    }
    kv {
      source => "rawSectionH"
      field_split => "\n"
      value_split => ":"
      target => "auditLogTrailer"
    }
`

**Comment by Alex-Kw** [comment]
I've submitted a pull request that works well enough for me. I will not be offended if you do not like it :)  I haven't been able to find a better way to contact you, but wanted to thank you for this project. I use it with mlogc and logstash http input plugin (behind nginx https) to ship logs to an elk stack from several remote sensors. It's been an awesome replacement to auditconsole so far. Cheers.

**Comment by bitsofinfo** [comment]
Should be addressed by 1.2.2
